content/posts/2023-12-31_constraining_complexity.markdown

```diff
@@ -0,0 +1,623 @@
+---
+date: "2023-12-31 10:52:59"
+title: "Constraining Complexity in the Generics Design"
+tags:
+- "programming"
+- "golang"
+summary: A supplementary post to my GopherConAU 2023 talk of the same title.
+---
+
+I gave a talk at [GopherConAU 2023](https://gophercon.com.au/) about a
+particular problem we encountered when designing generics for Go and what we
+might do about it.
+
+This blog post is meant as a supplement to that talk. It mostly reproduces its
+content, while giving some supplementary information and more detailed
+explanations where necessary.
+
+So if you prefer to ingest your information from text, than this blog post
+should serve you well. If you prefer a talk,
+[you can watch the recording](https://www.youtube.com/watch?v=dQw4w9WgXcQ)
+and use it to get some additional details in the relevant sections.
+
+The talk (and hence this post) is also a follow-up to
+[my earlier blog post about this](https://blog.merovius.de/posts/2022-05-16-calculating-type-sets/).
```

Comment:
I suggest replacing “this” with something more descriptive, e.g. “about type sets” or something like that